libtelemetry/src/main/java/com/mapbox/android/telemetry/CertificateBlacklist.java

```diff
@@ -0,0 +1,225 @@
+      }
+    }
+
+    return lastUpdateTime;
+  }
+
+  void updateBlacklist() {
+    HttpUrl requestUrl = new HttpUrl.Builder().scheme(HTTPS_SCHEME)
+      .host(determineConfigEndpoint())
+      .addPathSegment("events-config")
+      .addQueryParameter(ACCESS_TOKEN_QUERY_PARAMETER, accessToken)
+      .build();
+
+    Request request = new Request.Builder()
+      .url(requestUrl)
+      .build();
+
+    OkHttpClient client = new OkHttpClient();
+    client.newCall(request).enqueue(this);
+  }
+
+  private void saveBlackList(List<String> revokedKeys) {
+    String fileContents = createListContent(revokedKeys);
+    FileOutputStream outputStream;
+
+    try {
+      outputStream = context.openFileOutput(BLACKLIST_FILE, Context.MODE_PRIVATE);
+      outputStream.write(fileContents.getBytes());
+      outputStream.close();
+    } catch (Exception exception) {
+      logger.error(SAVE_BLACKLIST_FAIL, exception.getMessage());
+    }
+  }
+
+  @Override
+  public void onFailure(Call call, IOException exception) {
+    Log.e(LOG_TAG, REQUEST_FAIL, exception);
+  }
+
+  @Override
+  public void onResponse(Call call, Response response) throws IOException {
+    List<String> revokedKeys = extractResponse(response);
+
+    saveBlackList(revokedKeys);
+  }
+
+  private String createListContent(List<String> revokedKeys) {
+    Date date = new Date();
+
+    StringBuilder content = new StringBuilder(date.getTime() + NEW_LINE);
+
+    for (String key : revokedKeys) {
+      content.append(SHA256).append(key).append(NEW_LINE);
+    }
+
+    return content.toString();
+  }
+
+  private List<String> obtainBlacklistContents(File file) throws IOException {
+    InputStream inputStream = new FileInputStream(file);
+    BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream));
+
+    List<String> blacklist = new ArrayList<>();
+
+    try {
+      boolean done = false;
+      while (!done) {
+        String line = reader.readLine();
+        done = (line == null);
+
+        if (line != null && !line.isEmpty()) {
+          blacklist.add(line);
+        }
+      }
+    } catch (Exception exception) {
+      logger.error(READLINE_FAIL, exception.getMessage());
+    } finally {
+      reader.close();
+      inputStream.close();
+    }
+    return blacklist;
+  }
+
+  private String determineConfigEndpoint() {
+    EnvironmentChain environmentChain = new EnvironmentChain();
+    EnvironmentResolver setupChain = environmentChain.setup();
+    ServerInformation serverInformation;
+    try {
+      ApplicationInfo appInformation = context.getPackageManager().getApplicationInfo(context.getPackageName(),
+        PackageManager.GET_META_DATA);
+
+      if (appInformation != null && appInformation.metaData != null) {
+        serverInformation = setupChain.obtainServerInformation(appInformation.metaData);
+        return ENDPOINTS.get(serverInformation.getEnvironment());
+      }
+    } catch (Exception exception) {
```

Comment:
I wonder if it is possible not to use catch all exception here and be a little more explicit about error handling?